csrc/ir/base_nodes.h

```diff
@@ -585,6 +585,9 @@ class NVF_API Expr : public Statement {
   // TODO: Protect based on being in kernel container
   kir::Predicate* writePredicate() const;
 
+  // TODO: Protect based on being in kernel container
+  void setPredicate(kir::Predicate* predicate);
```

Comment:
Is this change necessary?

Reply:
reverted. Not required if change to use `withPredicate`